Authentication/AuthenticationCoordinator.swift

```diff
@@ -0,0 +1,232 @@
+         user selects that option.
+         
+         - Returns: A valid signup controller to use.
+    */
+    internal func createSignupController() -> SignupController {
+        return SignupController(configuration: createSignupControllerConfiguration())
+    }
+    
+    /**
+         Creates the signup view controller configuration
+         that the signup controller will use to access the
+         signup view model to use, the signup view to display
+         and the delegates for events and transitions to
+         other screens.
+     
+         It uses the components returned by the
+         components factory.
+     
+         - Returns: A valid signup controller configuration to use.
+     */
+    internal func createSignupControllerConfiguration() -> SignupControllerConfiguration {
+        let configuration = self._componentsFactory.createSignupViewConfiguration()
+        let createSignupView: () -> SignupViewType = { [unowned self] in
+            return self._componentsFactory
+                .createSignupView(self._componentsFactory
+                    .createSignupViewDelegate(configuration))
+        }
+        return SignupControllerConfiguration(
+            viewModel: _componentsFactory.createSignupViewModel(sessionService,
+                credentialsValidator: _componentsFactory.createSignUpCredentialsValidator(),
+                configuration: configuration),
+            viewFactory: createSignupView,
+            transitionDelegate: _componentsFactory.createSignupControllerTransitionDelegate() ?? self,
+            delegate: _componentsFactory.createSignupControllerDelegate(),
+            termsAndServicesURL: configuration.termsAndServicesURL)
+    }
+    
+    /**
+         Creates the terms and services controller
+         to use when the user selects that option.
+         
+         - Returns: A valid terms and services controller.
+     */
+    internal func createTermsAndServicesController(url: NSURL) -> TermsAndServicesController {
+        return TermsAndServicesController(url: url, delegate: _componentsFactory.createTermsAndServicesControllerDelegate())
+    }
+    
+}
+
+// MARK: - RecoverPassword Functions
+public extension AuthenticationCoordinator {
+    
+    /**
+         Creates the recover password main controller to use when the
+         user selects that option.
+         
+         - Returns: A valid recover password controller to use.
+     */
+    public func createRecoverPasswordController() -> RecoverPasswordController {
+        // TODO
+        return RecoverPasswordController()
+    }
+    
+}
+
+
+
+extension AuthenticationCoordinator: LoginControllerTransitionDelegate {
+    
+    public final func toSignup(controller: LoginController) {
+        if _componentsFactory.initialScreen == .Login {
+            let signupController = createSignupController()
+            controller.navigationController!.pushViewController(signupController, animated: true)
```

Comment:
In this case, we expect there to e a navigationcontroller because if not the app won't work as expected. So I think that using `!` is a good way avoid the situation of not working and not knowing why.